### PR TITLE
cache array lengths in OrderFulfiller for gas optimization

### LINKED ISSUE #337: Cache array length for gas optimization
Gas optimization, on [`OrderFulfiller.sol#L309`](https://github.com/ProjectOpenSea/seaport/blob/7aa48f09b00eeb3932d129796fd3141b2c6756d6/contracts/lib/OrderFulfiller.sol#L309) you could cache variable `orderParameters.offer.length` to save gas.

Pull request:
https://github.com/ProjectOpenSea/seaport/pull/336

### PR DESCRIPTION
**Pull request by ryanio**
Continues effort from #336

Closes #337

### CODE REVIEW

File: contracts/lib/OrderFulfiller.sol [hunk 1]
```diff
@@ -207,9 +207,11 @@ contract OrderFulfiller is
                 }
             }
 
+            // Cache offer length.
+            uint256 totalOffers = orderParameters.offer.length;
+
             // Iterate over each offer on the order.
-            // Skip overflow check as for loop is indexed starting at zero.
```
Comment (0age) [inline contracts/lib/OrderFulfiller.sol:211]: Let's keep this comment

File: contracts/lib/OrderFulfiller.sol [hunk 2]
```diff
@@ -207,9 +207,11 @@ contract OrderFulfiller is
                 }
             }
 
+            // Cache offer length.
```
Comment (0age) [inline contracts/lib/OrderFulfiller.sol:210]: "Read offer array length from memory and place on stack.

File: contracts/lib/OrderFulfiller.sol [hunk 3]
```diff
@@ -207,9 +207,11 @@ contract OrderFulfiller is
...
+            uint256 totalOffers = orderParameters.offer.length;
```
Comment (0age) [inline contracts/lib/OrderFulfiller.sol:211]: "totalOfferItems"

File: contracts/lib/OrderFulfiller.sol [hunk 4]
```diff
@@ -291,9 +293,12 @@ contract OrderFulfiller is
                 }
             }
 
+            // Cache consideration length.
+            uint256 totalConsiderations = orderParameters.consideration.length;
```
Comment (0age) [inline contracts/lib/OrderFulfiller.sol:297]: "totalConsiderationItems"

**Review by 0age — CHANGES_REQUESTED** [review]
Small nits, otherwise LGTM!

File: contracts/lib/OrderFulfiller.sol (hunk 1 above)
Comment (ryanio) [reply contracts/lib/OrderFulfiller.sol:211]: oops didn't mean to do that

**Pull request merged**